File: Day-01/clojure/DerTev/day1.clj
(ns day1
  (:require [clojure.string :as string]))

(defn get-elves-calories []
  (map #(reduce + %) (map (partial map read-string)
                          (map #(string/split % #"\r\n")
                               (-> (slurp "input.txt")
                                   (string/split #"\r\n\r\n"))))))

(defn part-one []
  (apply max (get-elves-calories)))

(defn part-two []
  (let [sorted (sort > (get-elves-calories))]
    (+ (nth sorted 0) (nth sorted 1) (nth sorted 2))))

(println (str "Part one: " (part-one)))
(println (str "Part two: " (part-two)))

File: Day-01/python/Matito/day1.py
f = open("day1_input.txt", "r").read().splitlines()

elves = []
calories = 0


for i in range(len(f)):

    if f[i]  == '':
        elves.append(calories)
        calories = 0
    else:
        calories = calories + int(f[i])

print(max(elves)) # part 1


elves.sort(reverse=True)
print(elves[0] + elves[1] + elves[2]) #part 2